sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Security/Ucfg/EntryPointRecognizer.cs

```diff
@@ -0,0 +1,55 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2018 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis;
+using SonarAnalyzer.Helpers;
+
+namespace SonarAnalyzer.Security.Ucfg
+{
+    public class EntryPointRecognizer
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Add a 'protected' constructor or the 'static' keyword to the class declaration. ([csharpsquid:S1118](https://sonarcloud.io/organizations/sonarsource/rules#rule_key=csharpsquid%3AS1118))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=sonaranalyzer-dotnet&pullRequest=1402&issues=AWNksP-Yneya_4cHgrMo&open=AWNksP-Yneya_4cHgrMo)

[ISSUE_KEY]: AWNksP-Yneya_4cHgrMo